certbot-dns-google/certbot_dns_google/_internal/tests/dns_google_test.py

```diff
@@ -88,6 +88,7 @@ class GoogleClientTest(unittest.TestCase):
     record_ttl = 42
     zone = "ZONE_ID"
     change = "an-id"
+    visibility = "public"
```

Comment:
One last thing and I think we are good to go. Could you craft a test (or edit one) where for given two zones for the same tld, one private and one public, the assertions validate that the plugin will take the public one ? You should put the private first in the list to ensure that the behavior is not driven by list ordering.

Reply:
Make sense @adferrand.

I have changed the evaluation behavior to get explicitly only zones with  `public` value on `visibility`. We don't expect changes on `forwarding` and `peering` DNS zones on `visibility` but I think it's better to get only the `public` ones.

I have also added the following [commit](https://github.com/certbot/certbot/pull/9744#:~:text=add%20test%20for%20split%20horizon%20dns%20google) tests for the Split Horizon DNS zones. I haven't found any test using `dnsName` on zone definition, so I have done the same using only the `id` and made the zone with `private` `visibility` appear first on the array.